internal/config/rtfiles.go

```diff
@@ -129,8 +129,15 @@ func AddRuntimeFilesFromAssets(fileType RTFiletype, directory, pattern string) {
 	if err != nil {
 		return
 	}
+
+assetLoop:
 	for _, f := range files {
 		if ok, _ := path.Match(pattern, f); ok {
+			for _, rf := range realFiles[fileType] {
+				if f == rf.Name()+strings.TrimPrefix(pattern, "*") {
```

Comment:
This assumes that the pattern is of the form `*.ext`, which may not always be the case. I see `AddRuntimeFilesFromAssets()` can be also called from lua, so the pattern may be arbitrary.

Can't we just do this?
```
if f.Name() == rf.Name() {
```

Reply:
IIRC file of user will return `<name>.<ext>` for `Name()` but file parsed from asset dir will return `<name>` for `Name()` so they won't be equal

Reply:
It's the other way around:
The realFile has no extension with `Name()`, but the namedFile has. The extension will be removed in the moment it's converted into the assetFile.
But at the end this doesn't matter, because we can remove the extension from the namedFile too and then this would fit in a comparison against the realFile.

This would be more generic then and we don't run into the troubles mentioned by @dmaluka.

Reply:
Or add extension to assetFile. I have a question can plugins rely on runtime files? Does micro provides to plugins method of getting runtime file? If yes, both addition or deletion extension can break some plugins that rely on it

Reply:
You both are more or less right. `f` in this context isn't one of each, it's an ordinary string containing the extension and thus we can't simply use `f.Name()`.
I updated it once again to convert the string into the `assetFile` before its comparison and this should now look more convenient.

Reply:
Yeah, the last version looks much better.